### ISSUE TITLE
Recommended best practices for EarlyStop and ReduceLROnPlateau callbacks with HyperBand tuner

### ISSUE THREAD

**Issue by romanovzky** [keras-team/keras-tuner#262]
Hi there,
Great package, I use it a lot and is now my main and go-to solution for model tuning in Keras.
I am wondering, what would be the best/recommended practices for LR and ES callbacks when using the HyperBand tuner?
Does the tuner evaluate the final loss at each bracket, or the best loss in the model history at each bracket? Does it even make sense to use these callbacks?
Thanks for the help!

**Comment by romanovzky** [comment]
By looking at the [MultiExecutionTuner](https://github.com/keras-team/keras-tuner/blob/e06bec6cac2172bd2cb038cca645578ef2ad307a/kerastuner/engine/multi_execution_tuner.py#L74) class, from which HyperBand inherits, it seems that it takes the best value of the objective
```python
model = self.hypermodel.build(trial.hyperparameters)
history = model.fit(*fit_args, **copied_fit_kwargs)
for metric, epoch_values in history.history.items():
    if self.oracle.objective.direction == 'min':
        best_value = np.min(epoch_values)
    else:
        best_value = np.max(epoch_values)
    metrics[metric].append(best_value)
```
if that is the case, then can I assume that using LR reducer and ES are good practices?

**Comment by borgarpa** [comment]
Hi! I am also very interested in this question, since I'm also user of the Hyperband algorithm.

I believe the best value is used in order to retrieve the optimal number of epochs. In that sense Early Stopping might be useful if you want to fasten the algorithm, although you might skip local optima. Furthermore, theoretically, Hyperband relies on allocating a certain proportion of the maximum resources provided (e.g., epochs). Reducing the total amount of epochs used (by early-stopping certain models) might cause the algorithm not to be thorough enough when searching for hyperparameter configurations.

Anyways, I am not sure about it, and I think it would be great if those techniques could be applied. It might as well speed up Hyperband, which would be fantastic. 

Does anyone know if they can be applied? EarlyStop and ReduceLROnPlateau would certainly be a great improvement in Hyperband's performance.

**Comment by syfantid** [comment]
Also interested in an answer. I've recently started using Hyperband and I noticed that in certain cases, while the training is in the final epochs, the model clearly overfits, so it'd benefit from early stopping. But I wouldn't want to mess with the inner workings of the algorithm.
